src/AbstractCsv.php

```diff
@@ -270,4 +270,30 @@ protected function setIterator()
 
         return new SplFileObject($this->getStreamFilterPath(), $this->open_mode);
     }
+
+    /**
+     * Returns the CSV document as a Generator of string chunk
+     *
+     * @param int $length number of bytes read
+     *
+     * @return Generator
+     */
+    public function chunk($length)
+    {
+        if ($length < 1) {
+            throw new InvalidArgumentException(sprintf('%s() expects the length to be a positive integer %d given', __METHOD__, $length));
+        }
+
+        $document = $this->getIterator();
+        $input_bom = $this->getInputBOM();
+        $document->rewind();
+        $document->fseek(strlen($input_bom));
+        foreach (str_split($this->output_bom.$document->fread($length), $length) as $chunk) {
+            yield $chunk;
+        }
+
+        while ($document->valid()) {
```

Comment:
@nyamsprod I just discovered that no method called `valid` exists on the object returned by `AbstractCsv::getIterator`. Is something like this an acceptable workaround?

```php
while (!$document->eof()) {
    // etc.
}
```

Reply:
Never mind, looks like `AbstractCsv::getIterator` always returns either `League\Csv\Modifier\StreamIterator`, or `SplFileObject`, both of which would allow calling `::valid` for this use case.